### PR TITLE
Apifirewall poc v2

### PR DESCRIPTION
**Pull request by colindomoney-42crunch**
Change(s): This is a package upgrade version of the 42Crunch API protection integration with Azure Sentinel. Upgrading to 2.0.3 and adding the package.

Reason for Change(s): Upgrading to 2.0.3 and adding the package.

Version Updated: This is version 2.0.3, and replaces 2.0.2.

Testing Completed: Yes, the solution has been built using V2 of the tool. The resultant output has been deployed in both the UI preview and deployment tool. Both work as expected. The deployment works exactly as expected and has been reviewed several times internally.

Checked that the validations are passing and have addressed any issues that are present: This pull request has been validated using both the KqlvalidationsTests and detectionTemplateSchemaValidation scripts, both return a pass result.

### CODE REVIEW

File: Solutions/42Crunch API Protection/Package/createUiDefinition.json
Comment (v-vdixit) [inline Solutions/42Crunch API Protection/Package/createUiDefinition.json:1]: Hi @colindomoney-42crunch The updates in create UI are not reflecting in accordance with the [latest tool for packaging](https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V2), could you please check and update the createUI file once, thanks!

Comment (colindomoney-42crunch) [reply Solutions/42Crunch API Protection/Package/createUiDefinition.json:1]: The only issue I can see is the incorrect logo path, I have corrected this to refer to the Azure-Sentil repository instead. Please confirm this is satisfactory.

Comment (v-vdixit) [reply Solutions/42Crunch API Protection/Package/createUiDefinition.json:1]: Hi @colindomoney-42crunch according to the latest tool after updating createUI the workbook tab should reflect data as shown in the following example:
image
Please check that the changes are inline with the latest tool update, if you need any further assistance please let me know, thanks!

Comment (colindomoney-42crunch) [reply Solutions/42Crunch API Protection/Package/createUiDefinition.json:1]: I have updated the description to provide detail of the 5 workbook elements provided by our plugin.

Comment (v-vdixit) [reply Solutions/42Crunch API Protection/Package/createUiDefinition.json:1]: Hi @colindomoney-42crunch, the workbook tab is still not reflecting the changes as per latest tool, 
image
Please use the latest tool provided at the given path - https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V2 - let me know if you have any queries related to this, thanks!

Comment (v-vdixit) [reply Solutions/42Crunch API Protection/Package/createUiDefinition.json:1]: Hi @colindomoney-42crunch we will complete reviewing this PR today, thanks

File: Solutions/42Crunch API Protection/Package/createUiDefinition.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
...
+        "description": "**Note:** _There may be [known issues](https://aka.ms/sentinelsolutionsknownissues) pertaining to this Solution, please refer to them before installing._\n\nThe 42Crunch API Protection solution protects APIs by installing a microfirewall inline with the API server. Access logs from the microfirewall are emitted to Microsoft Sentinel allowing analysis and investigation of attacks.\n\n**Data Connectors:** 1, **Workbooks:** 1, **Analytic Rules:** 11\n\n[Learn more about Microsoft Sentinel](https://aka.ms/azuresentinel) | [Learn more about Solutions](https://aka.ms/azuresentinelsolutionsdoc)",
```
Comment (v-vdixit) [inline Solutions/42Crunch API Protection/Package/createUiDefinition.json:9]: @colindomoney-42crunch Please add the logo path in the correct format in the description here to have the logo displayed with solution description, thanks!

File: Solutions/42Crunch API Protection/Data/Solution_42CrunchAPIProtection.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "Name": "42Crunch API Protection",
   "Author": "42Crunch - plugins@42crunch.com",
-  "Logo": "<img src=\"https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Logos/42CrunchLogo.svg\"width=\"75px\"height=\"75px\">",
+  "Logo": "<img src=\"https://raw.githubusercontent.com/42Crunch/azure-sentinel-integration/main/logo/42CrunchLogo.svg\"width=\"75px\"height=\"75px\">",
```
Comment (v-vdixit) [inline Solutions/42Crunch API Protection/Data/Solution_42CrunchAPIProtection.json:4]: Hi @colindomoney-42crunch please use the correct path for logo

File: Solutions/42Crunch API Protection/Package/2.0.3.zip
Comment (v-vdixit) [inline Solutions/42Crunch API Protection/Package/2.0.3.zip:1]: Hi @colindomoney-42crunch please keep the zip file inline with the latest changes in createUI and maintemplate, thanks!

Comment (colindomoney-42crunch) [reply Solutions/42Crunch API Protection/Package/2.0.3.zip:1]: Sure, I pushed before I built the package 😏

File: Solutions/42Crunch API Protection/Data/Solution_42CrunchAPIProtection.json (hunk 2 above)
Comment (colindomoney-42crunch) [reply Solutions/42Crunch API Protection/Data/Solution_42CrunchAPIProtection.json]: This is fixed, I had an old version of the solution file in my workspace.

File: Solutions/42Crunch API Protection/Package/createUiDefinition.json (hunk 1 above)
Comment (colindomoney-42crunch) [reply Solutions/42Crunch API Protection/Package/createUiDefinition.json]: This is fixed, I had an old version of the solution file in my workspace.

File: Solutions/42Crunch API Protection/Package/2.0.3.zip
Comment (v-vdixit) [inline Solutions/42Crunch API Protection/Package/2.0.3.zip:1]: Hi @colindomoney-42crunch, There should only be 'createUiDefinition' and 'mainTemplate' in zip file, please remove the other files and folders. Also, please ensure that the zip package has same files as the latest update in the solution, I can see some differences in the maintemplate in zip and the one outside, please check and update this, thanks!

Comment (colindomoney-42crunch) [reply Solutions/42Crunch API Protection/Package/2.0.3.zip:1]: I'm using the ./createSolutionV2.ps1 to build the package - I am not adding the previous Zip file. I will delete them before checking in.

Comment (colindomoney-42crunch) [reply Solutions/42Crunch API Protection/Package/2.0.3.zip:1]: I have both files in the Zip and they are exactly the same as in the base of the 'packages' folder. I have resubmitted now.

Comment (v-vdixit) [reply Solutions/42Crunch API Protection/Package/2.0.3.zip:1]: Thanks, will check and update

[the remark this answers is not in the document]
Comment (colindomoney-42crunch) [reply Solutions/42Crunch API Protection/Package/2.0.3.zip:1]: I don't understand - I have mainTemplate.json and createUiDefinition.json in 2.0.3.zip and the same two files in 2.0.2.zip. I am using the ./createSolutionV2.ps1 to produce the package as per instructions. I am at a loss to know what else I need to do - please could you explain precisely.

[the remark this answers is not in the document]
Comment (v-vdixit) [reply Solutions/42Crunch API Protection/Package/2.0.3.zip:1]: Hi @colindomoney-42crunch, while looking into the zip package folder it is expected to find createUIdefinition and maintemplate files but instead after opening it I can see this,
image
And after further opening the '2.0.3' folder I can see three files 
image

This package composition will not be acceptable while uploading the package on Azure Marketplace, so please check and update this, thanks!

[the remark this answers is not in the document]
Comment (colindomoney-42crunch) [reply Solutions/42Crunch API Protection/Package/2.0.3.zip:1]: Submitted the new Zip built on a Windows host - let's see if this is OK.

**Review by v-vdixit — APPROVED** [review]
This looks good, thanks!

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**